src/wallet/test/wallet_tests.cpp

```diff
@@ -329,6 +329,39 @@ BOOST_FIXTURE_TEST_CASE(importwallet_rescan, TestChain100Setup)
     }
 }
 
+// This test verifies that wallet settings can be added and removed
+// Concurrently, ensuring no race conditions occur during either process.
+BOOST_FIXTURE_TEST_CASE(write_wallet_settings_concurrently, TestingSetup)
+{
+    WalletContext context;
+    context.chain = m_node.chain.get();
+    const int NUM_WALLETS = 5;
+    std::vector<std::thread> threads;
+    {
```

Comment:
nit: brief docstrings to at-a-glance help see how the test is structured:

<details>
<summary>git diff on 135eb1d5b1</summary>

```diff
diff --git a/src/wallet/test/wallet_tests.cpp b/src/wallet/test/wallet_tests.cpp
index ec9792f7a9..d434c1c207 100644
--- a/src/wallet/test/wallet_tests.cpp
+++ b/src/wallet/test/wallet_tests.cpp
@@ -337,6 +337,7 @@ BOOST_FIXTURE_TEST_CASE(write_wallet_settings_concurrently, TestingSetup)
     context.chain = m_node.chain.get();
     const int NUM_WALLETS = 5;
     std::vector<std::thread> threads;
+    // Add NUM_WALLETS wallets concurrently, ensure we end up with NUM_WALLETS.
     {
         for (int i{0}; i < NUM_WALLETS; i++) {
             threads.emplace_back([i, &context] {
@@ -349,6 +350,7 @@ BOOST_FIXTURE_TEST_CASE(write_wallet_settings_concurrently, TestingSetup)
         BOOST_CHECK_EQUAL(wallets.getValues().size(), NUM_WALLETS);
     }
     threads.clear();
+    // Remove NUM_WALLETS wallets concurrently, ensure we end up with 0 wallets.
     {
         for (int i{0}; i < NUM_WALLETS; i++) {
             threads.emplace_back([i, &context] {

```
</details>

Reply:
Added